[CI] SamplingContextTests testScaling failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/3htvthdcoqyeg/tests/:server:test/org.elasticsearch.search.aggregations.support.SamplingContextTests/testScaling

**Reproduction line:**
`./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.support.SamplingContextTests.testScaling" -Dtests.seed=EF825246ED08FF2C -Dtests.locale=hi-IN -Dtests.timezone=Europe/Zagreb -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.aggregations.support.SamplingContextTests&tests.test=testScaling

**Failure excerpt:**
```
java.lang.AssertionError: 0.029171808509351443
Expected: (a value equal to or greater than <806110674907251L> and a value less than or equal to <806110674907251L>)
     but: a value equal to or greater than <806110674907251L> <806110674907238L> was less than <806110674907251L>

  at __randomizedtesting.SeedInfo.seed([EF825246ED08FF2C:C4C8FD16D20DEE36]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.elasticsearch.search.aggregations.support.SamplingContextTests.testScaling(SamplingContextTests.java:34)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/support/SamplingContextTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/support/SamplingContextTests.java
@@ -10,11 +10,8 @@ package org.elasticsearch.search.aggregations.support;
 
 import org.elasticsearch.test.ESTestCase;
 
-import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.closeTo;
 import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.greaterThanOrEqualTo;
-import static org.hamcrest.Matchers.lessThanOrEqualTo;
 
 public class SamplingContextTests extends ESTestCase {
     protected static final int NUMBER_OF_TEST_RUNS = 20;
@@ -31,7 +28,7 @@ public class SamplingContextTests extends ESTestCase {
             long rescaled = samplingContext.inverseScale(samplingContext.scale(randomLong));
             assertThat(
                 Double.toString(samplingContext.probability()),
-                (double)rescaled / randomLong,
+                (double) rescaled / randomLong,
                 // No matter how you scale `long` values, the inverse back may be a little off due to rounding
                 closeTo(1.0, 1e-4)
             );

--- a/server/src/test/java/org/elasticsearch/search/aggregations/support/SamplingContextTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/support/SamplingContextTests.java
@@ -20,22 +20,20 @@ public class SamplingContextTests extends ESTestCase {
     protected static final int NUMBER_OF_TEST_RUNS = 20;
 
     private static SamplingContext randomContext() {
-        return new SamplingContext(randomDoubleBetween(1e-8, 0.1, false), randomInt());
+        return new SamplingContext(randomDoubleBetween(1e-6, 0.1, false), randomInt());
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/83748")
     public void testScaling() {
         for (int i = 0; i < 20; i++) {
             SamplingContext samplingContext = randomContext();
-            long randomLong = randomLongBetween(100_000_000L, Long.MAX_VALUE);
+            long randomLong = randomLongBetween(1_000_000_000L, 100_000_000_000L);
             double randomDouble = randomDouble();
             long rescaled = samplingContext.inverseScale(samplingContext.scale(randomLong));
-            // No matter how you scale `long` values, the inverse back may be a little off
-            long error = (long) (rescaled * 1e-15);
             assertThat(
                 Double.toString(samplingContext.probability()),
-                rescaled,
-                allOf(greaterThanOrEqualTo(randomLong - error), lessThanOrEqualTo(randomLong + error))
+                (double)rescaled / randomLong,
+                // No matter how you scale `long` values, the inverse back may be a little off due to rounding
+                closeTo(1.0, 1e-4)
             );
             assertThat(
                 Double.toString(samplingContext.probability()),

